src/libs/NetworkConnection.ts

```diff
@@ -123,6 +124,13 @@ function subscribeToNetInfo(): () => void {
 
             // If a check is taking longer than this time we're considered offline
             reachabilityRequestTimeout: CONST.NETWORK.MAX_PENDING_TIME_MS,
+            reachabilityShouldRun: () => {
+                if (!hasPendingNetworkCheck) {
```

Comment:
I guess we can simplify this condition a bit:
`reachabilityShouldRun: () => !hasPendingNetworkCheck && (hasPendingNetworkCheck = true),`

Reply:
tbh I find the initial implementation provided easier to follow (less cognitively demanding).

What I'd think about though is limiting the nesting further more by introducing an early return (reverting the boolean check).

```diff
reachabilityShouldRun: () => {
-    if (!hasPendingNetworkCheck) {
-         hasPendingNetworkCheck = true;
-          return true;
-     }
-     return false;
+   if (hasPendingNetworkCheck) {
+     return false;
+   }
+
+   hasPendingNetworkCheck = true;
+   return true;
},
```